flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/Kafka09PartitionDiscoverer.java

```diff
@@ -69,12 +69,18 @@ protected void initializeConnections() {
 	}
 
 	@Override
-	protected List<KafkaTopicPartition> getAllPartitionsForTopics(List<String> topics) throws WakeupException {
+	protected List<KafkaTopicPartition> getAllPartitionsForTopics(List<String> topics) throws WakeupException, IllegalArgumentException {
 		List<KafkaTopicPartition> partitions = new LinkedList<>();
 
 		try {
 			for (String topic : topics) {
-				for (PartitionInfo partitionInfo : kafkaConsumer.partitionsFor(topic)) {
+				List<PartitionInfo> partitions = kafkaConsumer.partitionsFor(topic);
+
+				if (partitions == null) {
+					throw new IllegalArgumentException("Could not fetch partitions for Topic: %s".format(topic));
```

Comment:
`IllegalArgumentException` doesn't seem right here.
Maybe just a runtime exception?

Reply:
Also, would it make more sense to only fail if all topics fail to define partitions to consume from?

Reply:
I've changed it to a RuntimeException. My initial thought was to throw an IllegalArgumentException because the supplied topic didn't exist. I want to fail if one of the topics couldn't be retrieved, otherwise, it might fail silently. Also, this doesn't change the existing behavior. Let me know what you think.